### PR TITLE
Release 1.10.0 [8116]

### PR DESCRIPTION
**Pull request by MiguelCompany**
Bumping version and release notes

### CODE REVIEW

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/233/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/234/)

File: docs/notes.rst [hunk 1]
```diff
@@ -1,12 +1,62 @@
...
+* Solved issue where a publisher could block on write for a long time when, after a
+  large number of samples have been sent, a new subscriber is matched.
+* Correctly handling the case where lifespan expires at the same time on several samples.
+* Solved some issues regarding liveliness on writers with no readers.
+* Correctly removing changes from histories on keyed topics.
+* Not reusing cache change when sample does not fit.
+* Fixed custom wait_until methods when time is in the past.
+* Several data races and ABBA locks fixed.
+* Reduced CPU and memory usage.
+* Reduced flakyness of liveliness tests.
```
Comment (EduPonz) [inline docs/notes.rst:31]: I believe is flakiness. Modify `docs/spelling_wordlist.txt` accordingly

File: docs/notes.rst [hunk 2]
```diff
@@ -1,12 +1,62 @@
...
+
+It also includes the following bug fixes and improvements:
+
+* Fixed alignment on parameter lists
+* Fixed error sending more than 256 fragments.
+* Fix handling of STRICT_REALTIME.
+* Fixed submessage_size calculation on last data_frag.
+* Solved an issue when recreating a publishing participant with the same GUID.
+* Solved issue where a publisher could block on write for a long time when, after a
+  large number of samples have been sent, a new subscriber is matched.
```
Comment (EduPonz) [inline docs/notes.rst:23]: I took me 2-3 reads to understand the sentence, what about:
```suggestion
  * Solved issue where a publisher could block on write for a long time when a new subscriber (late joiner) is matched, if the publisher had already sent a large number of messages.
```

Comment (MiguelCompany) [reply docs/notes.rst:23]: 👍

File: docs/notes.rst (hunk 1 above)
Comment (MiguelCompany) [reply docs/notes.rst]: 👍

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/235/)

**Pull request merged**